src/ui/public/vislib/lib/handler/handler.js

```diff
@@ -116,14 +116,22 @@ export default function HandlerBaseClass(Private) {
       });
 
       // render the chart(s)
-      selection.selectAll('.chart')
-      .each(function (chartData) {
+      this.loadedCount = 0;
+      const chartSelection = selection.selectAll('.chart');
+      chartSelection.each(function (chartData) {
         const chart = new self.ChartClass(self, this, chartData);
 
         self.vis.activeEvents().forEach(function (event) {
           self.enable(event, chart);
         });
 
+        chart.events.on('rendered', () => {
+          self.loadedCount++;
+          if (self.loadedCount === chartSelection.length) {
+            charts[0].events.emit('renderComplete');
```

Comment:
Can you explain why this is using `charts[0]` in a comment in the code? I would expect this to happen at a higher level.